include/util/guidance/name_announcements.hpp

```diff
@@ -129,16 +131,34 @@ inline bool requiresNameAnnounced(const std::string &from_name,
 
     const auto pronunciation_changes = from_pronunciation != to_pronunciation;
 
-    return !obvious_change || needs_announce || pronunciation_changes;
+    // when exiting onto ramps, we need to be careful about exit numbers. These can often be only
+    // assigned to the first part of the ramp
+    //
+    //  a . . b . c . . d
+    //         ` e . . f
+    //
+    // could assign the exit number to `be` when exiting `abcd` instead of the full ramp.
+    //
+    // Issuing a new-name instruction here would result in the turn assuming the short segment to be
+    // irrelevant and remove the exit number in a collapse scenario. We don't want to issue any
+    // instruction from be-ef, since we only loose the exit number. So we want to make sure that we
+    // don't just loose an exit number, when exits change
+    const auto exits_change = from_exits != to_exits;
+    const auto looses_exit = (names_are_equal && !from_exits.empty() && to_exits.empty());
```

Comment:
The overall effect is 2), not announcing a new name. If we would announce a new name, the engine would see a short segment after which a new name is given and use the second name for the ramp instruction.
By not announcing a new name, we ensure that the name with the exit number is used for the ramp.